Tests/SwiftDocCTests/XCTestCase+LoadingTestData.swift

```diff
@@ -0,0 +1,79 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+@testable import SwiftDocC
+
+extension XCTestCase {
+    
+    /// Loads a documentation bundle from the given source URL and creates a documentation context.
+    func loadBundle(from bundleURL: URL, codeListings: [String : AttributedCodeListing] = [:], externalResolvers: [String: ExternalReferenceResolver] = [:], externalSymbolResolver: ExternalSymbolResolver? = nil, diagnosticFilterLevel: DiagnosticSeverity = .hint, configureContext: ((DocumentationContext) throws -> Void)? = nil) throws -> (URL, DocumentationBundle, DocumentationContext) {
+        let workspace = DocumentationWorkspace()
+        let context = try DocumentationContext(dataProvider: workspace, diagnosticEngine: DiagnosticEngine(filterLevel: diagnosticFilterLevel))
+        context.externalReferenceResolvers = externalResolvers
+        context.externalSymbolResolver = externalSymbolResolver
+        context.externalMetadata.diagnosticLevel = diagnosticFilterLevel
+        try configureContext?(context)
+        // Load the bundle using automatic discovery
+        let automaticDataProvider = try LocalFileSystemDataProvider(rootURL: bundleURL)
+        // Mutate the bundle to include the code listings, then apply to the workspace using a manual provider.
+        var bundle = try automaticDataProvider.bundles().first!
```

Comment:
```suggestion
        var bundle = try XCTUnwrap(automaticDataProvider.bundles().first)
```

Reply:
Nice! I didn't consider that these have access to all the test case functionality.